ui/app/AppLayouts/Profile/views/EnsView.qml

```diff
@@ -323,43 +323,52 @@ Item {
-            //% "Transaction pending..."
-            Global.toastMessage.title = qsTrId("ens-transaction-pending")
-            Global.toastMessage.source = Style.svg("loading")
-            Global.toastMessage.iconColor = Style.current.primary
-            Global.toastMessage.iconRotates = true
-            Global.toastMessage.link = `${ensView.ensUsernamesStore.getEtherscanLink()}/${txResult}`
-            Global.toastMessage.open()
+            let url = `${ensView.ensUsernamesStore.getEtherscanLink()}/${txResult}`;
+            Global.displayToastMessage(qsTr("Transaction pending..."),
+                                       "",
+                                       "",
+                                       true,
+                                       Constants.ephemeralNotificationType.normal,
+                                       url);
         }
         onTransactionCompleted: {
+            let title = ""
             switch(trxType){
-                case "RegisterENS":
-                    Global.toastMessage.title = !success ?
-                                         //% "ENS Registration failed"
-                                         qsTrId("ens-registration-failed")
-                                         :
-                                         //% "ENS Registration completed"
-                                         qsTrId("ens-registration-completed");
-                    break;
-                case "SetPubKey":
-                    Global.toastMessage.title = !success ?
-                                         //% "Updating ENS pubkey failed"
-                                         qsTrId("updating-ens-pubkey-failed")
-                                         :
-                                         //% "Updating ENS pubkey completed"
-                                         qsTrId("updating-ens-pubkey-completed");
-                    break;
+            case "RegisterENS":
+                title = !success ?
+                            //% "ENS Registration failed"
+                            qsTrId("ens-registration-failed")
+                          :
+                            //% "ENS Registration completed"
+                            qsTrId("ens-registration-completed");
+                break;
+            case "SetPubKey":
+                title = !success ?
+                            //% "Updating ENS pubkey failed"
+                            qsTrId("updating-ens-pubkey-failed")
+                          :
+                            //% "Updating ENS pubkey completed"
+                            qsTrId("updating-ens-pubkey-completed");
+                break;
+            default:
+                console.error("unknown transaction type: ", trxType);
+                return
+            }
+
+            let icon = "block-icon";
+            let ephType = Constants.ephemeralNotificationType.normal;
+            if (success) {
+                icon = "check-circle";
```

Comment:
Ternary operator suits here well.

Reply:
@endulab if we delve into details, with ternary operator we will have 2 exactly the same conditions one next to another, this way it's just one condition (one check).